### PR TITLE
feat: add sign in flow gf-7

### PR DESCRIPTION
**Pull request by perekotypole**
Implemented backend sign-in flow:

- add methods to find user in user service (by id and by email)
- /auth/sing-in returns user (without password) and token on correct email
- connect encryption service to compare passwords

### CODE REVIEW

File: apps/backend/src/modules/users/user.service.ts [hunk 1]
```diff
@@ -45,6 +57,19 @@ class UserService implements Service {
 		};
 	}
 
+	public async getByEmail(email: string): Promise<UserEntity> {
+		const item = await this.userRepository.findByEmail(email);
+
+		if (!item) {
+			throw new UserError({
+				message: "User doesn't exist",
```
[the remark this answers is not in the document]
Comment (perekotypole) [reply apps/backend/src/modules/users/user.service.ts:65]: Should it be common? Or is it better to specify it to user module?

[the remark this answers is not in the document]
Comment (liza-veis) [reply apps/backend/src/modules/users/user.service.ts:65]: We might have the same exception messages across different services or apps (backend/frontend). Usually, it is not directly linked to the service but more to the domain. It's not the only right way to do this, more like a stylistic question and that's the way we prefer in this project

File: packages/shared/src/modules/users/libs/exceptions/user-error.exception.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import {
+	type HTTPCode,
+	HTTPError,
+} from "../../../../libs/modules/http/http.js";
+import { type ValueOf } from "../../../../libs/types/value-of.type.js";
```
[the remark this answers is not in the document]
Comment (perekotypole) [reply packages/shared/src/modules/users/libs/exceptions/user-error.exception.ts:5]: We don't have typescript paths for shared, so we decided to leave this as it is

**Review by what1s1ove — APPROVED** [review]

**Review by yevhenii-hladkov — APPROVED** [review]

**Comment by perekotypole** [comment]
@liza-veis conflicts are fixed

File: apps/backend/src/modules/users/user.service.ts [hunk 3]
```diff
@@ -1,22 +1,25 @@
...
-import { type UserRepository } from "~/modules/users/user.repository.js";
-
-import { ExceptionMessage } from "./libs/enums/enums.js";
+} from "./libs/types/types.js";
+import { UserEntity } from "./user.entity.js";
+import { type UserRepository } from "./user.repository.js";
 
 class UserService implements Service {
-	private encryption: Encryption;
+	private encryption: BaseEncryption;
```
Comment (liza-veis) [inline apps/backend/src/modules/users/user.service.ts:16]: we should use `Encryption`, we need to export it from  `~/libs/modules/encryption/encryption.js`

```suggestion
	private encryption: Encryption;
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

**Review by liza-veis — APPROVED** [review]

**Pull request merged**